addons/stock/static/src/xml/forecast_widget.xml

```diff
@@ -1,12 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <templates id="template" xml:space="preserve">
     <t t-name="stock.forecastWidget">
-        <span t-if="['draft', 'partially_available', 'assigned'].includes(state)" t-esc="reserved_availability_str"/>
+        <span t-if="['draft', 'partially_available', 'assigned', 'cancel', 'done'].includes(state)" t-esc="reserved_availability_str"/>
         <span t-elif="!forecast_expected_date_str and will_be_fulfilled" class="text-success">Available</span>
         <span t-elif="forecast_expected_date_str and will_be_fulfilled" t-att-class="forecast_is_late ? 'text-danger' : 'text-warning'">Exp <t t-esc="forecast_expected_date_str"/></span>
         <span t-else="" class="text-danger">Not Available</span>
-        <button t-if="product_type == 'product'" class="o_forecast_report_button btn btn-link o_icon_button ml-2" title="Forecasted Report">
-            <i t-attf-class="fa fa-fw fa-area-chart {{ state != 'draft' and (!will_be_fulfilled or forecast_is_late) ? 'text-danger' : '' }}"/>
-        </button>
+        <t t-if="product_type == 'product'">
+            <button t-if="id" class="o_forecast_report_button btn btn-link o_icon_button ml-2" title="Forecasted Report">
+                <i t-attf-class="fa fa-fw fa-area-chart {{ state != 'draft' and (!will_be_fulfilled or forecast_is_late) ? 'text-danger' : '' }}"/>
+            </button>
+            <button t-else="" disabled="" class="o_forecast_report_button btn btn-link o_icon_button ml-2" title="Forecasted Report">
+                <i t-attf-class="fa fa-fw fa-area-chart {{ state != 'draft' and (!will_be_fulfilled or forecast_is_late) ? 'text-danger' : '' }}"/>
+            </button>
+        </t>
```

Comment:
```suggestion
        <button t-if="product_type == 'product'" t-att="[id ? 'enabled' : 'disabled', '']" class="o_forecast_report_button btn btn-link o_icon_button ml-2" title="Forecasted Report">
            <i t-attf-class="fa fa-fw fa-area-chart {{ state != 'draft' and (!will_be_fulfilled or forecast_is_late) ? 'text-danger' : '' }}"/>
        </button>
```

Reply:
Or with a `t-set` to avoid the `enabled` which isn't a real thing...
```suggestion
<t t-set="is_enabled" t-value="id ? False : ['disabled', '']"/>
        <button t-if="product_type == 'product'" t-attf="is_enabled" class="o_forecast_report_button btn btn-link o_icon_button ml-2" title="Forecasted Report">
            <i t-attf-class="fa fa-fw fa-area-chart {{ state != 'draft' and (!will_be_fulfilled or forecast_is_late) ? 'text-danger' : '' }}"/>
        </button>
</t>    
```